analyzers/src/SonarAnalyzer.Common/Rules/Utilities/SymbolReferenceAnalyzerBase.cs

```diff
@@ -112,33 +111,37 @@ protected sealed override SymbolReferenceInfo CreateMessage(SyntaxTree syntaxTre
                 var symbol => symbol
             };
 
-        private static SymbolReferenceInfo.Types.SymbolReference GetSymbolReference(List<ReferenceInfo> references, SyntaxTree tree)
+        private static SymbolReferenceInfo.Types.SymbolReference GetSymbolReference(IReadOnlyList<ReferenceInfo> references, string filePath)
         {
-            var declarationSpan = GetDeclarationSpan(references);
+            var declarationSpan = GetDeclarationSpan(references, filePath);
             if (!declarationSpan.HasValue)
             {
                 return null;
             }
 
-            var symbolReference = new SymbolReferenceInfo.Types.SymbolReference { Declaration = GetTextRange(Location.Create(tree, declarationSpan.Value).GetLineSpan()) };
+            var symbolReference = new SymbolReferenceInfo.Types.SymbolReference { Declaration = GetTextRange(declarationSpan.Value) };
             for (var i = 0; i < references.Count; i++)
             {
                 var reference = references[i];
-                if (!reference.IsDeclaration)
+                if (!reference.IsDeclaration
+                    && reference.Identifier.GetLocation().GetMappedLineSpanIfAvailable() is var mappedLineSpan
+                    && string.Equals(mappedLineSpan.Path, filePath, StringComparison.OrdinalIgnoreCase))
```

Comment:
For posterity, can we add a comment here, explaining why `mappedLineSpan.Path` and `filePath` can be different?

Just an inline `Not the case for generated code (e.g. razor)` or something like that.

There is no mention of Razor anywhere here around (the closest being in the `GetMappedLineSpanIfAvailable` method), and without context, it's hard to figure out how this condition can possibly be true.